### PR TITLE
feat: admin product list

### PR DESCRIPTION
**Pull request by likeri29**
I added the possibility of seeing the products with activity statuses (active and rejected) in the product list tab.
I added the possibility for the admin to delete products
[JIRA task](https://team-jfqe9wlknaql.atlassian.net/jira/software/c/projects/SB/boards/8?assignee=712020%3Abc32ac79-d2de-4a3e-accf-51396ebf73bb&selectedIssue=SB-173)

Screenshot 2024-07-11 at 16 11 19
Screenshot 2024-07-11 at 16 11 24

Screenshot 2024-07-11 at 17 11 28

### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet
### Frontend
- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the
- [x] No text in the components, use i18n approach
- [ ] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/pages/admin-panel/component/product-management/index.tsx [hunk 1]
```diff
@@ -138,13 +164,27 @@ const ProductManagement: React.FC = () => {
...
+				/>
 				{isLoading && <Loader />}
 				{isError && (
 					<CustomError
 						errorMessage={t("AdminProductManagement.errorLoadingProducts")}
 					/>
 				)}
+				{serverError && (
+					<Typography
+						color="error"
```
Comment (VladManukian) [inline src/pages/admin-panel/component/product-management/index.tsx]: color

File: src/redux/admin/admin-api.ts [hunk 2]
```diff
@@ -7,6 +7,12 @@ import { adminApiPath } from "./constants.ts";
 
 export const adminApi = api.injectEndpoints({
 	endpoints: (build) => ({
+		deleteProduct: build.mutation<void, string>({
+			query: (id) => ({
+				method: httpMethods.DELETE,
+				url: `${adminApiPath.DELETE_PRODUCT}/${id}`,
+			}),
+		}),
```
Comment (VladManukian) [inline src/redux/admin/admin-api.ts:15]: create test case for this endpoint pls

File: src/pages/admin-panel/component/modal/index.tsx
Comment (lyudad) [inline src/pages/admin-panel/component/modal/index.tsx:1]: better to create a reusable component for modal window which will accept needed params. delete popup will be used in all places where delete option exist

File: src/pages/admin-panel/component/product-management/index.tsx [hunk 3]
```diff
@@ -138,13 +164,27 @@ const ProductManagement: React.FC = () => {
...
 					<CustomError
 						errorMessage={t("AdminProductManagement.errorLoadingProducts")}
 					/>
 				)}
+				{serverError && (
+					<Typography
+						color={`${theme.palette.error}`}
+						sx={{ marginBottom: "8px" }}
+						variant="body2"
+					>
```
Comment (VladManukian) [inline src/pages/admin-panel/component/product-management/index.tsx:183]: Move element above to styles.ts or move all styles to sx.

**Review by lyudad — APPROVED** [review]

**Pull request merged**